core/domain/classifier/base_classification_algorithm.py

```diff
@@ -0,0 +1,92 @@
+# coding: utf-8
+#
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class of classification algorithms for free-form text answers"""
+
+from abc import abstractmethod
```

Comment:
For python standard libraries, do "import abc" and reference using abc.abstractmethod. (That said, see comment below.)